### DISCUSSION TITLE
Robot localisation and config file

### DISCUSSION THREAD

**Discussion by chaidop** [gstavrinos/ai_msc_robotic_systems_assignments#42] [Q&A]
Καλησπέρα και συγγνωμη αν πρόκειται για χαζή ερώτηση,
Προσπαθώ να τρέξω ενα αρχικο dummmy τροποποιημένο κώδικα της άσκησης αλλά έχω μια απορία.
Βλέπω από το rviz ότι το Mapping έχει γίνει ήδη καθώς αναπαράγεται το Map, το επόμενο βήμα είναι το localization στο οποίο και κολλάω. 
Πως μπορώ να δώσω στο ρομπότ τη θέση του στο χάρτη; Όσα κατάφερα να βρω στο ιντερνετ
έδιναν μόνο μέσω του rviz πατώντας στον χάρτη αλλά δε ξέρω πως θα το κάνω προγραμματιστικά.
Σκέφτομαι να πάρω το x,y,z από το TransformListener με το frame id base_link είναι σωστή προσέγγιση?
Μπορείς επίσης να μου θυμίσεις που είναι το αρχειάκι με το configurations του ρομπότ (αισθητήρες κλπ?)

Ευχαριστώ!

**Comment by gstavrinos** [comment]
Σου φαίνεται ότι το ρομπότ βρίσκεται σε λάθος θέση στο χάρτη; (Όχι, αλλά αν ναι, το rviz για να θέσει το pose στέλνει σε ένα topic. Αρκεί να βρεις αυτό το topic λοιπόν.)

Για να πάρεις τη θέση είναι αυτή μια λύση, αλλά δεν είναι η πιο σωστή. Ωστόσο θα μπορέσεις κανονικά να λύσεις ολόκληρη την εργασία με αυτή την προσέγγιση. (Μπορείς να δεις και αν το ίδιο το nav2 σου δίνει κάτι όσο το ρομπότ πλοηγείται.)

Σε πρώτη φάση﻿﻿ δεν θα χρειαστεί να πειράξεις κάτι από τις ρυθμίσεις του ρομποτ. Αν επιμένεις πως το χρειάζεσαι, τα πακέτα που σχετίζονται με το ρομπότ μας αρχίζουν με `pal_*`, `pmb2_*` και `tiago_*`, και βρίσκονται στο /opt/ros/humble/share/.

**Comment by Xindictus** [comment]
Σχετικά με τη θέση, έχω προσπαθήσει να την πάρω από ένα συγκεκριμένο `topic` το οποίο όμως στο `runtime` δεν μπαίνει ποτέ μέσα στην callback, ενώ αν κάνω `ros2 topic echo` παίρνω κανονικά πληροφορία. Έχω δημιουργήσει για το `subscription` ένα `QoS` profile που να κάνει match το αντίστοιχο του `publisher` σε περίπτωση που του δημιουργούσε πρόβλημα, αλλά και πάλι τίποτα.

Πώς μπορώ να κάνω debug σε αυτή τη περίπτωση να δω τι φταίει;

**Reply by gstavrinos** [reply]
Το πρώτο πράγμα που θα δοκίμαζα θα ήταν να δω αν κώδικας μπλοκάρει κάπου πριν τη δημιουργία του subscription. Φαντάζομαι ότι έχεις ένα print ή κάτι παρόμοιο μέσα στην callback, οπότε κάτι παρόμοιο πριν και μετά το subscription για να δεις ότι όλα αρχικοποιούνται σωστά, είναι το πρώτο βήμα. (Ίσως αξίζει δικό της thread η ερώτηση)
